ui_components/src/village/Village.js

```diff
@@ -0,0 +1,265 @@
+import { apiFetch } from "../utils/network.js";
+
+function Village({
+    playerSeat,
+    villageName,
+    villageAPI,
+    policyData,
+    populationData,
+    seatData,
+    pointsData,
+    diplomacyData,
+    resourceData,
+    clanData,
+}) {
+    const [seatDataState, setSeatDataState] = React.useState(seatData);
+    const [resourceDataState, setResourceDataState] = React.useState(resourceData);
+    const [playerSeatState, setPlayerSeatState] = React.useState(playerSeat);
+    const resourceDays = React.useRef(1);
+    const resourceDaysDisplay = React.useRef("daily");
+    const FetchResources = () => {
+        apiFetch(
+            villageAPI,
+            {
+                request: 'LoadResourceData',
+                days: resourceDays.current,
+            }
+        ).then((response) => {
+            if (response.errors.length) {
+                handleErrors(response.errors);
+                return;
+            }
+            switch (resourceDays.current) {
+                case 1:
+                    resourceDays.current = 7;
```

Comment:
Little confused, why do we change to 7 if the current value is 1?

Reply:
On-click it toggled between Daily => Weekly => Monthly => Daily

Reply:
Hmm, could we handle that with a wrapper around this function, or rename this to `FetchNextIntervalTypeResources` to be clear on what it's doing?